packages/flutter/test/material/autocomplete_test.dart

```diff
@@ -254,6 +254,104 @@ void main() {
+          return kOptions.where((String option) {
+            return option.contains(textEditingValue.text.toLowerCase());
+          });
+        },
+      ),
+    )));
+
+    final Finder listFinder = find.byType(ListView);
+    final Finder inputFinder = find.byType(TextFormField);
+    await tester.tap(inputFinder);
+    await tester.enterText(inputFinder, '');
+    await tester.pump();
+    final Size baseSize = tester.getSize(listFinder);
+    final double resultingHeight = baseSize.height;
+    expect(resultingHeight, equals(200));
+  });
+
+  testWidgets('the options height restricts to max desired height', (WidgetTester tester) async {
+    const double desiredHeight = 150.0;
+    await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+      body: Autocomplete<String>(
+        optionsBuilder: (TextEditingValue textEditingValue) {
+          return kOptions.where((String option) {
+            return option.contains(textEditingValue.text.toLowerCase());
+          });
+        },
+        optionsMaxHeight: desiredHeight,
+      ),
+    )));
+
+    /// entering "a" returns 9 items from kOptions so basically the
+    /// height of 9 options would be beyond `desiredHeight=150`,
+    /// so height gets restricted to desiredHeight.
+    final Finder listFinder = find.byType(ListView);
+    final Finder inputFinder = find.byType(TextFormField);
+    await tester.tap(inputFinder);
+    await tester.enterText(inputFinder, 'a');
+    await tester.pump();
+    final Size baseSize = tester.getSize(listFinder);
+    final double resultingHeight = baseSize.height;
+
+    /// expected desired Height =150.0
+    expect(resultingHeight, equals(desiredHeight));
+  });
+
+  testWidgets('The height of options shrinks to height of resulting items, if less than maxHeight', (WidgetTester tester) async {
+    // Returns a Future with the height of the default [Autocomplete] options widget
+    // after the provided text had been entered into the [Autocomplete] field.
+    Future<double> _getDefaultOptionsHeight(
+        WidgetTester tester, String enteredText) async {
+      final Finder listFinder = find.byType(ListView);
+      final Finder inputFinder = find.byType(TextFormField);
+      final TextFormField field = inputFinder.evaluate().first.widget as TextFormField;
+      field.controller!.clear();
+      await tester.tap(inputFinder);
+      await tester.enterText(inputFinder, enteredText);
+      await tester.pump();
+      final Size baseSize = tester.getSize(listFinder);
+      return baseSize.height;
+    }
+
+    const double maxOptionsHeight = 250.0;
+    await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+      body: Autocomplete<String>(
+        optionsBuilder: (TextEditingValue textEditingValue) {
+          return kOptions.where((String option) {
+            return option.contains(textEditingValue.text.toLowerCase());
+          });
+        },
+        optionsMaxHeight: maxOptionsHeight,
+      ),
+    )));
```

Comment:
indentations are off. we also tend to leave `child` as the last parameter to prevent trailing parameters in a list of parameters, and `optionsBuilder` is sort of the equivalent here.
```suggestion
    await tester.pumpWidget(MaterialApp(
      home: Scaffold(
      body: Autocomplete<String>(
        optionsMaxHeight: maxOptionsHeight,      
        optionsBuilder: (TextEditingValue textEditingValue) {
          return kOptions.where((String option) {
            return option.contains(textEditingValue.text.toLowerCase());
          });
        },
      ),
    )));
```